kubert/src/runtime.rs

```diff
@@ -59,12 +69,42 @@ pub struct Runtime<S = NoServer> {
     server: S,
     #[cfg(not(feature = "server"))]
     server: std::marker::PhantomData<S>,
+
+    #[cfg(feature = "prometheus-client")]
+    metrics: Option<ResourceWatchMetrics>,
 }
 
 /// Indicates that no HTTPS server is configured
 #[derive(Debug, Default)]
 pub struct NoServer(());
 
+#[cfg(feature = "prometheus-client")]
+#[derive(Clone, Debug, Hash, PartialEq, Eq, EncodeLabelSet)]
+struct ResourceWatchLabels {
+    kind: String,
+    group: String,
+    version: String,
+}
+
+#[cfg(feature = "prometheus-client")]
+#[derive(Clone, Debug, Hash, PartialEq, Eq, EncodeLabelSet)]
+struct ResourceWatchErrorLabels {
+    kind: String,
+    group: String,
+    version: String,
+    error: String,
```

Comment:
```suggestion
    error: &'static str,
```

Since all of our error types are known statically, we don't need to allocate to encode them.

Alternatively, you could add a ResourceWatchError enum that implements EncodeLabelValue.
```suggestion
    error: ResourceWatchError,
```